person.rb

```diff
@@ -0,0 +1,28 @@
+class Person
+  def initialize(age, name = nil, parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name || 'unknown'
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+
+  attr_accessor :name, :age
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  public
+
+  def can_use_services?
+    @age >= 18 || @parent_permission == true
```

Comment:
- [ ] Please use the private method 'of_age?' here to check a person is valid to use services rather than repeating yourself.

Reply:
Thanks